delfin/drivers/dell_emc/power_store/rest_handler.py

```diff
@@ -0,0 +1,958 @@
+        return used_capacity
+
+    def get_disks(self, storage_id):
+        disk_list = []
+        hardware_list = self.rest_call(self.REST_HARDWARE_URL)
+        for hardware in hardware_list:
+            lifecycle_state = hardware.get('lifecycle_state')
+            if consts.CHARACTER_DRIVE != hardware.get('type') or \
+                    lifecycle_state == consts.CHARACTER_EMPTY:
+                continue
+            extra_details = hardware.get('extra_details')
+            capacity = ''
+            firmware = ''
+            physical_type = constants.DiskPhysicalType.UNKNOWN
+            if extra_details:
+                firmware = extra_details.get('firmware_version')
+                drive_type = extra_details.get('drive_type')
+                if drive_type in consts.DiskType.ALL:
+                    continue
+                physical_type = consts.DISK_PHYSICAL_TYPE.get(
+                    drive_type, constants.DiskPhysicalType.UNKNOWN)
+                capacity = extra_details.get('size')
+            disk_result = {
+                'name': hardware.get('name'),
+                'storage_id': storage_id,
+                'native_disk_id': hardware.get('id'),
+                'serial_number': hardware.get('serial_number'),
+                'manufacturer': 'DELL EMC',
+                'firmware': firmware,
+                'capacity': capacity,
+                'status': consts.DISK_STATUS_MAP.get(
+                    lifecycle_state, constants.DiskStatus.NORMAL),
+                'physical_type': physical_type,
+                'logical_type': constants.DiskLogicalType.UNKNOWN,
+                'location': str(hardware.get('slot'))
+            }
+            disk_list.append(disk_result)
+        return disk_list
+
+    def get_controllers(self, storage_id):
+        list_controllers = []
+        nodes = self.get_node()
+        ips = self.get_ip()
+        hardware_list = self.rest_call(self.REST_HARDWARE_URL)
+        for hardware in hardware_list:
+            lifecycle_state = hardware.get('lifecycle_state')
+            if consts.CHARACTER_NODE != hardware.get('type') or \
+                    lifecycle_state == consts.CHARACTER_EMPTY:
+                continue
+            slot = hardware.get('slot')
+            appliance_id = hardware.get('appliance_id')
+            node_id = nodes.get('{}{}'.format(appliance_id, slot), {})
+            address = ips.get('{}{}'.format(appliance_id, node_id))
+            if not address:
+                LOG.error('mgmt_ip is empty, Exceptions may occur in snmptrap')
+            extra_details = hardware.get('extra_details')
+            memory_size = ''
+            cpu_info = ''
+            if extra_details:
+                memory_size = extra_details.get(
+                    'physical_memory_size_gb', 0) * units.Gi
+                cpu_info = extra_details.get('cpu_model')
+            full_name = hardware.get('name')
+            name = ''
+            if full_name:
```

Comment:
If `full_name` is None, we should use `location` as the name of the controller, not an empty string.

Reply:
The location doesn't necessarily have data, but the id definitely does

controller_id = hardware.get('id')
            controller_result = {
                'name': name if name else controller_id,